Gems/PhysX/Code/Tests/EditorMeshColliderComponentTests.cpp

```diff
@@ -147,30 +147,285 @@ namespace PhysXEditorTests
 
         EntityPtr gameEntity = CreateActiveGameEntityFromEditorEntity(editorEntity.get());
 
-        // since there was an editor rigid body component, the runtime entity should have a dynamic rigid body
-        const auto* rigidBody = gameEntity->FindComponent<PhysX::RigidBodyComponent>()->GetRigidBody();
-        const auto* pxRigidDynamic = static_cast<const physx::PxRigidDynamic*>(rigidBody->GetNativePointer());
+        // check that the runtime entity has the expected components
+        auto* meshColliderComponent = gameEntity->FindComponent<PhysX::MeshColliderComponent>();
+        auto* staticRigidBodyComponent = gameEntity->FindComponent<PhysX::StaticRigidBodyComponent>();
+        EXPECT_TRUE(meshColliderComponent != nullptr);
+        EXPECT_TRUE(staticRigidBodyComponent != nullptr);
 
-        PHYSX_SCENE_READ_LOCK(pxRigidDynamic->getScene());
+        AzPhysics::ShapeColliderPairList shapeConfigList = meshColliderComponent->GetShapeConfigurations();
+        EXPECT_EQ(shapeConfigList.size(), 1);
+
+        for (const auto& shapeConfigPair : shapeConfigList)
+        {
+            EXPECT_EQ(shapeConfigPair.second->GetShapeType(), Physics::ShapeType::PhysicsAsset);
+        }
+
+        const auto* simulatedBody = staticRigidBodyComponent->GetSimulatedBody();
+        const auto* pxRigidStatic = static_cast<const physx::PxRigidStatic*>(simulatedBody->GetNativePointer());
+
+        PHYSX_SCENE_READ_LOCK(pxRigidStatic->getScene());
 
         // there should be a single shape on the rigid body and it should be a box
-        EXPECT_EQ(pxRigidDynamic->getNbShapes(), 1);
-        if (pxRigidDynamic->getNbShapes() > 0)
+        EXPECT_EQ(pxRigidStatic->getNbShapes(), 1);
+        if (pxRigidStatic->getNbShapes() > 0)
         {
             physx::PxShape* shape = nullptr;
-            pxRigidDynamic->getShapes(&shape, 1, 0);
+            pxRigidStatic->getShapes(&shape, 1, 0);
             EXPECT_EQ(shape->getGeometryType(), physx::PxGeometryType::eSPHERE);
         }
 
+        const AZ::Aabb aabb = simulatedBody->GetAabb();
+        EXPECT_THAT(aabb.GetMin(), UnitTest::IsCloseTolerance(AZ::Vector3(-0.5f, -0.5f, -0.5f), 1e-3f));
+        EXPECT_THAT(aabb.GetMax(), UnitTest::IsCloseTolerance(AZ::Vector3(0.5f, 0.5f, 0.5f), 1e-3f));
+
+        delete meshAssetData;
+    }
+
+    // [GHI-14907]
+    // Asset Scale (with non-uniform value) does not work when the asset contains primitives and there
+    // is no non-uniform scale component present.
+    TEST_F(PhysXEditorFixture, DISABLED_EditorMeshColliderComponent_AssetWithPrimitive_AssetScale_CorrectShapeTypeGeometryTypeAndAabb)
+    {
+        auto* meshAssetData = AZ::Utils::LoadObjectFromBuffer<PhysX::Pipeline::MeshAssetData>(
+            PhysXMeshTestData::SpherePrimitive.data(), PhysXMeshTestData::SpherePrimitive.size());
+        EXPECT_TRUE(meshAssetData != nullptr);
+        if (!meshAssetData)
+        {
+            return;
+        }
```

Comment:
```suggestion
        ASSERT_TRUE(meshAssetData != nullptr);
```

http://google.github.io/googletest/reference/assertions.html#EXPECT_TRUE

Reply:
will change all these. Thanks.